UEBPrep/app/models/user.rb

```diff
@@ -62,6 +62,7 @@ def self.find_for_twitter_oauth(auth, signed_in_resource=nil)
 
         user = User.create(provider:auth.provider,
                            uid:auth.uid,
+                           username:auth.uid+"@twitter.com",
                            email:auth.uid+"@twitter.com",
```

Comment:
@caesarfish awesome, seems like we're on the same wavelength on this one. After you push your changes and verify it works, merge this branch at your discretion.